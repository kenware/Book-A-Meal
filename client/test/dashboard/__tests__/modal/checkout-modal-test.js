import React from 'react';
import { shallow } from 'enzyme';
import CheckOutModal from '../../../../components/dashboard/modals/checkOutModal';
import { props, emptyProps } from '../../dashboardMockData';

describe('Test checkout modal', () => {
  it('renders correctly with empty props', () => {
    const tree = shallow(<CheckOutModal {...emptyProps} onChange={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render with props', () => {
    const tree = shallow(<CheckOutModal {...props} onChange={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });
});
