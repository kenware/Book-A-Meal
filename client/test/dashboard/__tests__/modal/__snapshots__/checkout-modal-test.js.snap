// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test checkout modal renders correctly with empty props 1`] = `
<div
  className="modal-wrapper top-margin"
>
  <p
    className="order-center"
  >
    Cart
  </p>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text address"
    >
       Address: 
    </span>
    <textarea
      className="modal-menu "
      id="address"
      name="address"
      onChange={[MockFunction]}
      rows="4"
      style={
        Object {
          "marginLeft": "1rem",
        }
      }
      type="text"
      value=""
    />
  </div>
  <div
    className=""
  >
    <h3
      className="danger justify l-r-pad-text"
    />
  </div>
  <p
    className="order-center"
  >
    Cart Empty
  </p>
</div>
`;

exports[`Test checkout modal should render with props 1`] = `
<div
  className="modal-wrapper top-margin"
>
  <p
    className="order-center"
  >
    Cart
  </p>
  <div
    className="modal-contents"
  >
    <span
      className="modal-items l-r-pad-text address"
    >
       Address: 
    </span>
    <textarea
      className="modal-menu "
      id="address"
      name="address"
      onChange={[MockFunction]}
      rows="4"
      style={
        Object {
          "marginLeft": "1rem",
        }
      }
      type="text"
      value="Reveend street"
    />
  </div>
  <div
    className=""
  >
    <h3
      className="danger justify l-r-pad-text"
    >
      error occure
    </h3>
    error occure
  </div>
  <div>
    <table
      className="table"
    >
      <tbody>
        <tr
          className="p-white tr-color tr-height"
        >
          <th>
            S/N
          </th>
          <th>
            Name
          </th>
          <td>
            Quantity
          </td>
          <td>
            Remove
          </td>
        </tr>
        <tr
          className="order-contents tr tr-height"
          key="rice"
        >
          <td>
            1
          </td>
          <td>
            rice
          </td>
          <td>
            3
          </td>
          <td>
            <em
              className="fa fa-trash-o"
              id="cart-btn"
              onClick={[Function]}
              role="button"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
